### DISCUSSION TITLE
How can I substitute text in selected section instead of the whole line?

### DISCUSSION THREAD

**Discussion by sytdas** [neovim/neovim#30989] [Q&A] [answered]
I need to do some mapping between API data and python code. 
```py
class MyClass:
  x: XType  # actual name of field x
  y: YType  # actual name of field y
```

This is what I want to achieve 

```py
class MyClass:
  actual_name_of_field_x: XType  # actual name of field x
  actual_name_of_field_y: YType  # actual name of field y
```
I want to do this with macro so I can apply to other fields as well. But after copying the comment to beginning I stuck with replacing spaces with _.

**Comment by fitrh** [comment]
After copying the comment to the beginning

```python
class MyClass:
  actual name of field x: XType  # actual name of field x
  actual name of field y: YType  # actual name of field y
```

You can use `:h sub-replae-expression` with `:h substitute()`

```vim
:s/\w\+.*:/\=substitute(submatch(0), " ", "_", "g")
```

**Comment by telemachus** [comment, marked as answer]
Another idea is [`:help \%V`](https://neovim.io/doc/user/pattern.html#%2F%5C%25V). I.e., if you paste `actual name of field x` over `x`, then you can use `:'<,'>s/\%V /_/` to get the result that you want. (You can reselect manually before the substitution using `gv`, but you don't even need to.)
